From khali@linux-fr.org Wed Nov  2 15:35:52 2005
Date: Wed, 2 Nov 2005 21:42:48 +0100
From: Jean Delvare <khali@linux-fr.org>
To: Greg KH <greg@kroah.com>
Subject: i2c: writing-client doc update complement
Message-Id: <20051102214248.29ef2124.khali@linux-fr.org>
Content-Disposition: inline; filename=i2c-doc-writing-clients-fix-2.patch

My latest update to the writing-clients i2c documentation file was
incomplete, here's the complement.

Large parts of this file are still way out-of-date, but at least now
the memory allocation and freeing instructions are consistent.

Signed-off-by: Jean Delvare <khali@linux-fr.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 Documentation/i2c/writing-clients |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- gregkh-2.6.orig/Documentation/i2c/writing-clients
+++ gregkh-2.6/Documentation/i2c/writing-clients
@@ -412,7 +412,7 @@ For now, you can ignore the `flags' para
         release_region(address,FOO_EXTENT);
     /* SENSORS ONLY END */
     ERROR1:
-      kfree(new_client);
+      kfree(data);
     ERROR0:
       return err;
   }
@@ -443,7 +443,7 @@ much simpler than the attachment code, f
       release_region(client->addr,LM78_EXTENT);
     /* HYBRID SENSORS CHIP ONLY END */
 
-    kfree(data);
+    kfree(i2c_get_clientdata(client));
     return 0;
   }
 
